lib/api/playlist.js

```diff
@@ -0,0 +1,182 @@
+const ApiResource = require('./api_resource')
+const PageableApiResource = require('./pageable_api_resource')
+const qs = require('qs')
+
+/**
+ * @class Channel
+ */
+class Playlist extends ApiResource {
+  /**
+   * Retrieves flag if playlist is available for a specific channel.
+   *
+   * @param {Number}    channelId
+   *
+   * @returns {*}
+   */
+  isEnabled(channelId) {
+    return new Promise((resolve, reject) => {
+      this.context.authRequest('get', `/channels/${channelId}/settings/playlists.json`).then((res) => {
+        resolve(res.is_enabled)
+      }).catch((err) => {
+        reject(err)
+      })
+    })
+  }
+
+  /**
+   * Lists all Playlists on an channel.
+   *
+   * @param {Number} pageSize - The number of results to show per page.
+   * @param {Number} page     - The page to retrieve.
+   * @returns {Promise}
+   */
+  list(channelId, pageSize = 50, page = 1, filter = 'include_empty') {
+    return new Promise((resolve, reject) => {
+      /**
+       * @var {{channels, paging}} res
+       */
+      this.context.authRequest('get', `/channels/${channelId}/playlists.json?pagesize=${pageSize}&page=${page}&filter=${filter}`)
+        .then((res) => {
+          resolve(new PageableApiResource(this.context, 'playlists', res.playlists, res.paging))
+        })
+        .catch((err) => {
+          try {
+            if (err.response.status === 404) {
+              resolve(new PageableApiResource(this.context, 'playlists', {}, {
```

Comment:
Intercepting the 404 response and replacing it with a value can be misleading. The non-erroneous result implies to external programs that this resource exists, even though it does not.

I haven't looked into the playlist API in great detail (yet). Is there a reason why 404's must be handled internally?